packs/nest/actions/fan.yaml

```diff
@@ -0,0 +1,24 @@
+---
+name: fan
+runner_type: run-python
+description: Manage fan state
+enabled: true
+entry_point: fan.py
+parameters:
+  state:
+    type: string
+    description: State to set the fan
+    enum:
+      - "on"
+      - auto
+    default: auto
+    required: true
+  structure:
+    type: integer
+    description: ID of Nest Home name
+    default: 0
+  device:
```

Comment:
I also played with this pattern a few ways last night. A lot of them were to reduce the amount of code duplication (as a direct limitation to my knowledge of Python).

First option: function defaults. Minimal code, but hard-coded and does not give the user an easy way to globally override the default Structure/Device IDs.
Second option: Metaparameter defaults. No code, but lots of duplication in each file. Could drift
Third option: Global defaults. One stop shop, but logic overhead in each action to check if a value has been passed. But, tightly binds script to StackStorm via `config.txt`. I could abstract this a bit, but feels like it breaks the spirit of the action.

I sort of landed on the second option in both terms of speed and ease to write. Having a python action managing whether or not a default parameter or not seemed ugly, and something I wanted to bring up. I feel like this data binding should happen for free somewhere and have a data-binding convention on a per-action basis. This keeps the actions themselves focused on the lifting, and not logic to interpret and bootstrap the action.